deltadb/src/deltadb_expr.c

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright (C) 2015- The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file COPYING for details.
+*/
+
+#include "jx.h"
+
+#include <string.h>
+#include <stdio.h>
+
+struct deltadb_expr {
+	char operator[32];
+	char *param;
+	char *val;
+	struct deltadb_expr *next;
+};
+
+struct deltadb_expr * deltadb_expr_create( const char *str, struct deltadb_expr *next )
+{
+	struct deltadb_expr *e = malloc(sizeof(*e));
+	e->param = strdup(str);
+	char *delim = strpbrk(e->param, "<>=!");
+	e->val = strpbrk(delim, "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ");
+	int operator_size = (int)(e->val-delim);
+	strncpy(e->operator,delim,operator_size);
+	e->operator[operator_size] = '\0';
+	delim[0] = '\0';
+	e->next = next;
+	return e;
+}
+
+static int is_number(char const* p)
+{
+	char* end;
+	strtod(p, &end);
+	return !*end;
```

Comment:
It think here you want return end != p.

From the man page, if no conversion is performed, end is set to p.